datajoint/schema.py

```diff
@@ -151,15 +152,26 @@ def __call__(self, cls):
 
         if issubclass(cls, Part):
             raise DataJointError('The schema decorator should not be applied to Part relations')
-        self.process_relation_class(cls, context=self.context)
+        ext = {
+            cls.__name__: cls,
+            'self': cls
+        }
+        self.process_relation_class(cls, context=dict(self.context, **ext))
+
         # Process part relations
-        for part in cls._ordered_class_members:
+        for part in dir(cls):
```

Comment:
the reason we used `_ordered_class_members` is that they are ordered the same way as in the definition.  Thus parts that are defined in a specific order are processed in the same order.  This is important because parts may refer to one another.  Using `dir` will break that.